vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssSpecificationSpecModel.kt

```diff
@@ -0,0 +1,314 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.eclipse.kuksa.vssprocessor.spec
+
+import com.google.devtools.ksp.processing.KSPLogger
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.ParameterSpec
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.plusParameter
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.STAR
+import com.squareup.kotlinpoet.TypeName
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.kotlinpoet.asTypeName
+import org.eclipse.kuksa.vsscore.model.VssNode
+import org.eclipse.kuksa.vsscore.model.VssProperty
+import org.eclipse.kuksa.vsscore.model.VssSpecification
+import org.eclipse.kuksa.vsscore.model.className
+import org.eclipse.kuksa.vsscore.model.name
+import org.eclipse.kuksa.vsscore.model.variableName
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+import kotlin.reflect.full.declaredMemberProperties
+
+// Reflects the specification file as a model and is filled via reflection. That is why the variable names
+// should exactly match the names inside the specification file and be of a string type.
+internal class VssSpecificationSpecModel(
+    override var uuid: String = "",
+    override var vssPath: String = "",
+    override var description: String = "",
+    override var type: String = "",
+    override var comment: String = "",
+    @Suppress("MemberVisibilityCanBePrivate") var datatype: String = "",
+) : VssSpecification, SpecModel {
+    var childSpecifications = mutableListOf<VssSpecificationSpecModel>()
+
+    private lateinit var logger: KSPLogger
+    private lateinit var packageName: String
+
+    private val datatypeProperty: TypeName
+        get() {
+            return when (datatype) {
+                "string" -> String::class.asTypeName()
+                "boolean" -> Boolean::class.asTypeName()
+                // Do not use UInt because it is incompatible with @JvmOverloads annotation
+                "uint8", "uint16", "uint32" -> Int::class.asTypeName()
+                "int8", "int16", "int32" -> Int::class.asTypeName()
```

Comment:
DataType may also theoretically consist of int64 and uint64 which are internally handled as longs, see Types:2272 and Types:3214